### PR TITLE
secretize: don't secretize already secret func args

### PR DESCRIPTION
**Pull request by asraa**
Fixes https://github.com/google/heir/issues/809

### CODE REVIEW

File: tests/secretize/idempotent.mlir [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// RUN: heir-opt -secretize -wrap-generic -secretize %s | FileCheck %s
```
Comment (AlexanderViand-Intel) [inline tests/secretize/idempotent.mlir:1]: Maybe make the test a bit more explicit, as a change/bug in wrap-generic could cause it to regress to double-wrapping even for non-doubled-up `secret<secret<types>>`
```suggestion
// RUN: heir-opt -secretize -wrap-generic -secretize -wrap-generic %s | FileCheck %s
```

**Review by AlexanderViand-Intel — APPROVED** [review]
Thanks for fixing this so quickly! Just a tiny (optional) suggestion:

File: tests/secretize/idempotent.mlir (hunk 1 above)
Comment (asraa) [reply tests/secretize/idempotent.mlir]: nice! done. also added a check to ensure that there's only one secret.generic in that case

**Review by MeronZerihun — COMMENTED** [review]
LGTM :)

**Pull request merged**